manifests/init.pp

```diff
@@ -127,6 +127,7 @@
   Array $proxy_hide_header                                   = [],
   Array $proxy_pass_header                                   = [],
   Array $proxy_ignore_header                                 = [],
+  Array $resolver                                            = [],
```

Comment:
Should this be a `Array[String]` or can we even narrow it down to `Array[Stdlib::IP::Address]`?

Reply:
I currently use strings to be able to add extra settings for "valid" and "ipv6" to the end of the directive, but that could (and probably should) be handled better. I can add a couple of extra params for controlling those instead which would allow us to put proper validation on the resolver array.

Reply:
We could start with a string and later make it `Variant[Stdlib::IP::Address, Struct[...]]`. I also realized we might need to escape the IPv6 addresses which you could do in a string yourself.

Reply:
On second thought... the parameter should support address:port combinations if we are trying to match the nginx feature but stdlib doesn't seem to have a way to validate that combo. Any thoughts on how to handle that? I still think adding the extra params is better, but I don't believe doing so will allow us to narrow down the validation at all.

Reply:
You could use data types:

```puppet
type Nginx::Resolver = Variant[Array[Nginx::Address], Struct[
  ips               => Array[Nginx::Address],
  Optional['ipv6']  => Enum['on', 'off'],
  Optional['valid'] => Nginx::Time,
]]
type Nginx::Time = String // TODO: stricter validation
type Nginx::Address = Variant[
  Stdlib::IP::Address,
  Struct[
    ip   => Stdlib::IP::Address,
    port => Stdlib::Port,
  ],
]
```

We may want to allow simple arrays as well. The template to output this would be somewhat complicated though so you probably want to recast it in puppet code:
```puppet
if $resolver =~ Array {
  $_resolver = { 'ips' => $resolver }
} else {
  $_resolver = $resolver
}
```